src/components/MaterialUI/DataDisplay/TransferList.js

```diff
@@ -0,0 +1,203 @@
+		fontSize: theme.spacing(1.3),
+		lineHeight: theme.spacing(1.4),
+		padding: theme.spacing(0.7),
+		minWidth: theme.spacing(8),
+	},
+}));
+
+const useListStyles = makeStyles(theme => ({
+	item: {
+		fontSize: theme.spacing(1.3),
+		color: theme.palette.text.primary,
+		fontFamily: theme.typography.fontFamily,
+		padding: theme.spacing(0, 1.6),
+		cursor: "pointer",
+	},
+	selectedItem: {
+		backgroundColor: theme.palette.primary.light,
+	},
+	itemTitle: {
+		fontWeight: theme.typography.fontWeightSemiBold,
+	},
+}));
+
+const formatState = (state, checked, from, to) => {
+	const fromSelectedItems = state[from].filter(({ id }) => checked.includes(id));
+	return {
+		[to]: [...state[to], ...fromSelectedItems],
+		[from]: state[from].filter(({ id }) => !checked.includes(id)),
+	};
+};
+
+export const CustomList = ({ items, checked, setChecked, listItemFormatter }) => {
+	const classes = useListStyles();
+
+	const handleToggle = useCallback(
+		value =>
+			setChecked(state => {
+				const currentIndex = state.indexOf(value);
+				const newChecked = [...state];
+
+				if (currentIndex === -1) {
+					newChecked.push(value);
+				} else {
+					newChecked.splice(currentIndex, 1);
+				}
+				return newChecked;
+			}),
+		[setChecked],
+	);
+
+	return (
+		<List component="div" role="list">
+			{items.map(item => (
+				<ListItem
+					className={classNames(classes.item, checked.includes(item.id) && classes.selectedItem)}
+					key={item.id}
+					role="listitem"
+					onClick={!item.disabled ? () => handleToggle(item.id) : null}
+					disabled={item.disabled}
+				>
+					{listItemFormatter ? (
+						listItemFormatter(item)
+					) : (
+						<ListItemText
+							key={`transfer-list-item-${item.id}-label`}
+							primary={<Typography className={classes.itemTitle}>{item.title}</Typography>}
+							secondary={<Typography>{item.subtitle}</Typography>}
+						/>
+					)}
+				</ListItem>
+			))}
+		</List>
+	);
+};
+
+const TransferList = ({
+	rightListData,
+	leftListData,
+	onChange,
+	addButtonTitle,
+	removeButtonTitle,
+	customLeftComponent,
+	listItemFormatter,
+	height = 48,
+}) => {
+	const classes = useStyles({ height });
+	const [checked, setChecked] = useState([]);
+	const onAddButtonClick = () => {
+		onChange(formatState({ right: rightListData.data, left: leftListData.data }, checked, "left", "right"));
+		setChecked(checked.filter(check => !leftListData.data?.some(({ id }) => id === check)));
+	};
+	const onRemoveButtonClick = () => {
+		onChange(formatState({ right: rightListData.data, left: leftListData.data }, checked, "right", "left"));
+		setChecked(checked.filter(check => !rightListData.data?.some(({ id }) => id === check)));
+	};
+
+	const rightSelected = rightListData.data?.some(({ id }) => checked.includes(id));
+	const leftSelected = leftListData.data?.some(({ id }) => checked.includes(id));
+
+	return (
+		<Grid container spacing={1} justify="center" alignItems="center" className={classes.root}>
```

Comment:
Some time ago I realized that using spacing property on Grids without external wrapper (like div) could cause to some undesirable margins because of how that spacing property is implemented (it adds negative margin to a whole grid container).

So, what I would suggest is to wrap Grid container on the top in div just in case to avoid any issues in future.